File: readability/readability.py
from cs50 import get_string
from sys import exit

user_ip = get_string("Text: ")
ip_len = len(user_ip)


def main():

    if ip_len == 0:
        exit(1)

    num_lets = let_count()
    num_words = word_count()
    num_sents = sent_count()
    average = 1.0 / num_words * 100

    print_grade(num_lets, num_sents, average)

    # COMPLETE!!!
    exit(0)


# print compute average and print results
def print_grade(num_lets, num_sents, average):
    avg_lets = num_lets * average
    avg_sents = num_sents * average

    index = round(0.0588 * avg_lets - 0.296 * avg_sents - 15.8)

    if index < 1:
        print("Before Grade 1")
    elif index >= 16:
        print("Grade 16+")
    else:
        print("Grade", index)


# count number of sentences
def sent_count():
    count = 0

    for c in user_ip:
        if c == "." or c == "!" or c == "?":
            count += 1

    return count


# count number of words
def word_count():
    # +1 for word at end of sentence that has no space
    count = 1
    for c in user_ip:
        if c.isspace():
            count += 1

    return count


# count total number of letters
def let_count():
    count = 0

    for c in user_ip:
        if c.isalpha():
            count += 1

    return count


# call main() even though order of functions are not right
if __name__ == "__main__":
    main()

File: readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

//prototypes of functions
int letter_count(string user_input, int input_len);
int word_count(string user_input, int input_len);
int sentence_count(string user_input, int input_len);
void print_grade(int num_letters, int num_sentences);

float AVG = 0;

int main(void)
{
    string user_input = get_string("Text: ");
    int input_len = strlen(user_input);
    //  printf("input_len = %i\n", input_len);

    if (input_len == 0)
    {
        return 1;
    }

    int num_letters = letter_count(user_input, input_len);
    int num_words = word_count(user_input, input_len);
    int num_sentences = sentence_count(user_input, input_len);

    AVG = 1.0 / num_words * 100;
    //  printf("avg = %f\n", avg);

    print_grade(num_letters, num_sentences);
}


void print_grade(int num_letters, int num_sentences)
{
    // printf("----- in index() -----\n");

    float avg_letters = num_letters * AVG;
    // printf("avg_letters = %f\n", avg_letters);
    float avg_sentences = num_sentences * AVG;
    // printf("avg_sentences = %f\n", avg_sentences);

    int index = round(0.0588 * avg_letters - 0.296 * avg_sentences - 15.8);
    // printf("index = %i\n", index);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");

    }
    else
    {
        printf("Grade %i\n", index);
    }
}


int sentence_count(string user_input, int input_len)
{
    // printf("----- in sentence_count() -----\n");

    int counter = 0;
    int len = input_len;
    string input = user_input;

    for (int i = 0; i < len; i++)
    {
        if (input[i] == 33 || input[i] == 46 || input[i] == 63)
        {
            counter++;
        }
    }

    // printf("counter: %i\n", counter);
    return counter;
}


int word_count(string user_input, int input_len)
{
    // printf("----- in word_count() -----\n");

    // +1 for word at the end of the sentence that has no space
    int counter = 1;
    int len = input_len;
    string input = user_input;

    for (int i = 0; i < len; i++)
    {
        if (isspace(input[i]))
        {
            counter++;
        }
    }

    // printf("counter: %i\n", counter);
    return counter;
}


int letter_count(string user_input, int input_len)
{
    // printf("----- in letter_count() -----\n");

    int counter = 0;
    int len = input_len;
    string input = user_input;

    for (int i = 0; i < len; i++)
    {
        if (isalpha(input[i]))
        {
            counter++;
        }
    }

    // printf("counter: %i\n", counter);
    return counter;
}